packages/block-editor/src/components/block-list/index.js

```diff
@@ -201,6 +201,11 @@ function Items( {
 						getDefaultBlockName(),
 						rootClientId
 					) );
+			const hasSelectedRoot = !! (
+				rootClientId &&
+				selectedBlockClientId &&
+				rootClientId === selectedBlockClientId
+			);
```

Comment:
The previous code relied on returned type semantics; this ensures that client IDs are checked for equality.